app/services/api/teachers/school_transfers/query.rb

```diff
@@ -0,0 +1,172 @@
+module API::Teachers::SchoolTransfers
+  class Query
+    include Queries::FilterIgnorable
+
+    attr_reader :scope
+
+    def initialize(
+      lead_provider_id:,
+      updated_since: :ignore,
+      sort: { created_at: :asc }
+    )
+      @scope = Teacher.distinct
+
+      where_lead_provider_is(lead_provider_id)
+      where_updated_since(updated_since)
+      set_sort_by(sort)
+    end
+
+    def school_transfers
+      preload_associations(block_given? ? yield(scope) : scope)
+    end
+
+  private
+
+    def preload_associations(results)
+      results
+        .strict_loading
+        .includes(
+          ect_at_school_periods: {
+            earliest_training_period: {
+              school_partnership: :school,
+              active_lead_provider: :lead_provider
+            },
+            latest_training_period: {
+              school_partnership: :school,
+              active_lead_provider: :lead_provider
+            }
+          },
+          mentor_at_school_periods: {
+            earliest_training_period: {
+              school_partnership: :school,
+              active_lead_provider: :lead_provider
+            },
+            latest_training_period: {
+              school_partnership: :school,
+              active_lead_provider: :lead_provider
+            }
+          }
+        )
+    end
+
+    def where_lead_provider_is(lead_provider_id)
```

Comment:
I appretiate this is still WIP so you may be planning it anyway, but is it worth extracting some of this into methods to DRY it up and make the overall logic easier to follow?